### ISSUE TITLE
Remove rule QueryLogicalOrAndLogicalAndToArrayParameterRector

### ISSUE THREAD

**Issue by froemken** [sabbelasichon/typo3-rector#3071]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version |  0.12.23             |

**Issue**

Please remove rule `QueryLogicalOrAndLogicalAndToArrayParameterRector`

Rector want's me to update my code as follows:

```
     {
         $query = $this->createQuery();
         $query->matching(
-            $query->logicalOr(
-                $query->like('name', '%' . trim($name) . '%'),
-                $query->equals('warn_cell_id', $name)
-            )
+            $query->logicalOr([$query->like('name', '%' . trim($name) . '%'), $query->equals('warn_cell_id', $name)])
         );
         return $query->execute()->toArray();
     }
```

Applied rules:
`QueryLogicalOrAndLogicalAndToArrayParameterRector (https://docs.typo3.org/m/typo3/book-extbasefluid/master/en-us/6-Persistence/3-implement-individual-database-queries.html)`

The given link does not contain any `logicalAnd` or `logicalOr` containing an array.

Further you will find following comment in `logicalAnd` and `logicalOr` in Extbase source code:

```
/*
 * todo: Deprecate accepting an array as $constraint1
 *       Add param type hints for $constraint1 and $constraint2
 *       Make $constraint2 mandatory
 *       Add AndInterface return type hint
 *       Adjust method signature in interface
 */
```

So, please removve rule `QueryLogicalOrAndLogicalAndToArrayParameterRector` from rector

Stefan

**Comment by helsner** [comment]
Relates: #2462

**Comment by helsner** [comment]
closed due to reasoning found in PR #3072 
Thanks for the contribution!

**Comment by simonschaufi** [comment]
One can always exclude single rules with:

```php
$rectorConfig->services()->remove(QueryLogicalOrAndLogicalAndToArrayParameterRector::class);
```
